src/frontend/src/pages/index.js

```diff
@@ -91,20 +91,21 @@ export default function IndexPage() {
   }
 
   const getPostsCount = useCallback(async () => {
-    if (update) {
-      try {
-        const res = await fetch(`${telescopeUrl}/posts`, { method: 'HEAD' });
-        if (!res.ok) {
-          throw new Error(res.statusText);
-        }
-        setUpdate(false);
-        return res.headers.get('x-total-count');
-      } catch (error) {
-        console.log(error);
+    if (!shouldCheckForNewPosts) {
+      return null;
+    }
+    try {
+      const res = await fetch(`${telescopeUrl}/posts`, { method: 'HEAD' });
+      if (!res.ok) {
+        throw new Error(res.statusText);
       }
+      setShouldCheckForNewPosts(false);
```

Comment:
If we error on line 100 above, we won't set this to `false`.  Is that OK?  Should we do this in a `finally` block, or you want it to remain `true?

Reply:
You're right, a `finally` block is required to set it to `false`